v4/newrelic/application.go

```diff
@@ -5,26 +5,36 @@ package newrelic
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	"go.opentelemetry.io/otel/api/global"
 	"go.opentelemetry.io/otel/api/propagation"
 	"go.opentelemetry.io/otel/api/trace"
+
+	"github.com/newrelic/go-agent/v4/internal/logger"
 )
 
 // Application represents your application.  All methods on Application are nil
 // safe.  Therefore, a nil Application pointer can be safely used as a mock.
 type Application struct {
 	tracer      trace.Tracer
 	propagators propagation.Propagators
+	logger      Logger
 }
 
 // StartTransaction begins a Transaction with the given name.
 func (app *Application) StartTransaction(name string) *Transaction {
 	if app == nil {
 		return nil
 	}
+	app.initLogger()
```

Comment:
This should be in `NewApplication`.

Reply:
That was where I had it originally, but tests were failing if someone just creates an app manually - this test: